tarodb/asset_minting.go

```diff
@@ -266,6 +268,7 @@ func (a *AssetMintingStore) CommitMintingBatch(ctx context.Context,
 				EmissionEnabled: seedling.EnableEmission,
 			})
 			if err != nil {
+				err = castSqlError(err)
```

Comment:
So instead of cast, I think we should consider this approach: https://github.com/lightninglabs/taro/pull/190#discussion_r1023406169

In this case, we'd check the error outside this closure, and return a nicer error back to the caller. Something like `tarogarden.ErrDuplicateAssetName` or something like that.